### PR TITLE
Extended test for full beam search export

### PR DESCRIPTION
**Pull request by jamesr66a**
Since I wasn't running on a devserver, I substituted in a directory containing local copies of some checkpoint files for `check_dir`. I substituted the original path back in before pushing.

Note we found a bug in handling of shapes with vocab reduction, which is fixed here: https://github.com/pytorch/pytorch/pull/7303

pytorch:caffe2 comparison passes with the given tolerances, same as the other tests

### CODE REVIEW

File: pytorch_translate/test/extended_test_onnx.py [hunk 1]
```diff
@@ -346,3 +347,81 @@ def test_batched_beam_from_checkpoints_vr(self):
...
+                atol=1e-6,
+            )
+
+        beam_search.save_to_db(
+            os.path.join(tmp_dir, 'beam_search.predictor_export'),
+        )
+
+    def test_beam_search(self):
+        check_dir = (
+            '/mnt/gfsdataswarm-global/namespaces/search/language-technology-mt/'
```
Comment (jmp84) [inline pytorch_translate/test/extended_test_onnx.py]: This is not runnable externally.

Comment (jamesr66a) [reply pytorch_translate/test/extended_test_onnx.py]: ....neither is anything else in this file

Comment (jmp84) [reply pytorch_translate/test/extended_test_onnx.py]: Yep, this file will get deleted soon, I would just send it through diff instead if that's ok.

**Comment by jmp84** [comment]
@jamesr66a, could you send this one internally.

**Pull request closed without merging**